src/Compilers/CSharp/Test/Emit2/Semantics/CollectionExpressionTests.cs

```diff
@@ -9991,72 +9990,3081 @@ static void Main()
-                """
-                IOperation:  (OperationKind.None, Type: T[]) (Syntax: '[a, b]')
-                  Children(2):
-                      IParameterReferenceOperation: a (OperationKind.ParameterReference, Type: T, IsImplicit) (Syntax: 'a')
-                      IParameterReferenceOperation: b (OperationKind.ParameterReference, Type: T, IsImplicit) (Syntax: 'b')
+            var verifier = CompileAndVerify(
+                new[] { source, s_collectionExtensionsWithSpan },
+                targetFramework: targetFramework,
+                verify: Verification.Skipped,
+                expectedOutput: IncludeExpectedOutput("[1, 2], [3, null], "));
+            verifier.VerifyIL("Program.F1<T>", """
+                {
+                    // Code size       28 (0x1c)
+                    .maxstack  4
+                    IL_0000:  ldc.i4.2
+                    IL_0001:  newarr     "T"
+                    IL_0006:  dup
+                    IL_0007:  ldc.i4.0
+                    IL_0008:  ldarg.0
+                    IL_0009:  stelem     "T"
+                    IL_000e:  dup
+                    IL_000f:  ldc.i4.1
+                    IL_0010:  ldarg.1
+                    IL_0011:  stelem     "T"
+                    IL_0016:  call       "System.Span<T> System.Span<T>.op_Implicit(T[])"
+                    IL_001b:  ret
+                }
                 """);
-
-            var tree = comp.SyntaxTrees[0];
+            verifier.VerifyIL("Program.F2<T>", """
+                {
+                  // Code size       28 (0x1c)
+                  .maxstack  4
+                  IL_0000:  ldc.i4.2
+                  IL_0001:  newarr     "T"
+                  IL_0006:  dup
+                  IL_0007:  ldc.i4.0
+                  IL_0008:  ldarg.0
+                  IL_0009:  stelem     "T"
+                  IL_000e:  dup
+                  IL_000f:  ldc.i4.1
+                  IL_0010:  ldarg.1
+                  IL_0011:  stelem     "T"
+                  IL_0016:  call       "System.ReadOnlySpan<T> System.ReadOnlySpan<T>.op_Implicit(T[])"
+                  IL_001b:  ret
+                }
+                """);
+        }
+
+        [Fact]
+        public void RefSafety_Return_04()
+        {
+            string source = """
+                using System;
+                delegate Span<T> D<T>();
+                class Program
+                {
+                    static void Main()
+                    {
+                        D<int> d = () => [1, 2, 3];
+                        Span<int> s = d();
+                    }
+                }
+                """;
+            var comp = CreateCompilation(source, targetFramework: TargetFramework.Net80);
+            comp.VerifyEmitDiagnostics(
+                // (7,26): error CS9203: A collection expression of type 'Span<int>' cannot be used in this context because it may be exposed outside of the current scope.
```

Comment:
Perhaps the diagnostic message should say something like `exposed outside of the current function`.